pkg/apis/options/providers.go

```diff
@@ -76,6 +76,10 @@ type Provider struct {
 	Scope string `json:"scope,omitempty"`
 	// AllowedGroups is a list of restrict logins to members of this group
 	AllowedGroups []string `json:"allowedGroups,omitempty"`
+	// Code challenge methods supported by the Provider
+	SupportedCodeChallengeMethods []string `json:"code_challenge_methods_supported,omitempty"`
```

Comment:
Do we need this? If the user is manually configuring the code challenge, surely they just need to set `CodeChallengeMethod`?

Reply:
So this will be populated where? I would expect to see this in the actual provider, but not in the public config if this is for future discovery work?

Reply:
It's populated right after you scan the discovery document and fill codeChallengeAlgs in providers.go

I think I still need it to that I can compare the provider config vs the flag arguments.

Reply:
Nothing should be adding to the configuration while things are being constructed. I would expect this to be populated somewhere in the provider data construction right? In which case, this should be a member of the provider data, not of the providers configuration